cheroot/ssl/builtin.py

```diff
@@ -76,7 +123,8 @@ def wrap(self, sock):
                                     server_side=True, certfile=self.certificate,
                                     keyfile=self.private_key,
                                     ssl_version=ssl.PROTOCOL_SSLv23,
-                                    ca_certs=self.certificate_chain)
+                                    ca_certs=self.certificate_chain,
```

Comment:
I'm pretty sure based on other code I've seen that `ssl.wrap_socket` is the legacy way to handle SSL and that `context.wrap_socket` is the preferred way to handle it. The whole reason the concept of the `context` was added was to avoid scattering SSL parameters as attributes of every class that touches sockets. Instead, the context object encapsulates these parameters.

If that's true (and I'm not certain it is), any support for SSL cert validation should be provided first on self.context and then only if absolutely necessary, also provide legacy support. Perhaps the context object is sufficient to satisfy all modern deployments.

In fact, I'd  be tempted to say that the legacy technique for wrapping a socket should soon be deprecated and removed in favor of the context model.

Still, I don't think that eliminates the value of this PR. This PR could indeed adapt itself to demonstrate and test the context-based SSL certificatate validation support, but at the same time avoid the added complexity of additional pass-through parameters (with interpretation).

What do you think of that approach?

Reply:
@dpeschman

Reply:
> I'd be tempted to say that the legacy technique for wrapping a socket should soon be deprecated and removed in favor of the context model.

I have so many things on my mind that I hadn't connected that only a few days prior, I'd done just that in e4be9d6 (pending review in #60).